src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,23 @@
+package duke.task;
+
+public class Deadline extends Task {
+
+    protected String byDate;
+
+    public Deadline(String description, String byDate) {
+        super(description);
+        this.byDate=byDate;
+    }
+
+    public String getTypeIcon(){
+        return "[D]";
```

Comment:
Note usage of magic literals/constants, perhaps declare it at the top and return that constant instead.